### ISSUE TITLE
Randomize AI parameters in multiplayer to create personality (GNA #16006)

### ISSUE THREAD

**Issue by wesnoth-bugs** [wesnoth/wesnoth#1211]
**[Original submission](https://gna.org/bugs/index.php?16006)** by seahen on 2010-05-08

Wesnoth could provide more challenge, variety and replay value if you got a different set of AI opponents, each with a unique style of play, each time you played a non-campaign multiplayer game. Looking through http://www.wesnoth.org/wiki/AiWML I see the following parameters that could be randomized:

aggression = -1..0.75 (average 0.5)

attack_depth = 2..5

caution = 0.1..1.0 (average 0.25)

grouping = 70% offensive, 15% defensive, 15% no

leader_aggression = -8..0

leader_value = 1..5

number_of_possible_recruits_to_force_recruit = 2..5

protect_leader = 0.5..3.5

protect_leader_radius = 5..15

recruitment_pattern = (include each type 1..3 times)

scout_village_targeting = 1.5..4.5

village_value = 0.5..1.5

villages_per_scout = 2..6

This would be especially useful on random maps, where the map hasn't been specifically designed for the AI's default play style.

(Reproduced on all)
Release: trunk
Priority: 5 - Normal
Severity: 3 - Normal

**Comment by sevu** [comment]
Sounds like an idea for an add-on.

**Comment by cooljeanius** [comment]
The AI parameters available have changed, so some of these are no longer things, but I still think it's a good idea overall.

**Comment by CelticMinstrel** [comment]
Hmm. I don't think there _is_ any way of setting randomized values for aspects…

**Comment by cooljeanius** [comment]
> Hmm. I don't think there _is_ any way of setting randomized values for aspects…

It could probably be done in an event using the `[modify_ai]` tag, even if it can't be done in the initial declaration of `[ai]` parameters under the `[side]` tag... (at least, that's how I was thinking of doing it in a version of IftU where I'd mod it to make the Chaos faction actually behave chaotically...)

**Comment by CelticMinstrel** [comment]
Yes, that is possible, of course. The other thing I thought of was defining the aspects as Lua aspects, though I'm not quite sure if that can be made to produce a single random value that persists unchanged over the whole scenario.

**Comment by cooljeanius** [comment]
Related: #6319
